doc/source/rllib/single-agent-episode.rst

```diff
@@ -0,0 +1,214 @@
+extract this information inside their custom :py:class:`~ray.rllib.connectors.connector_v2.ConnectorV2`
+pipeline and add the data to the neural network batch.
+
+Another advantage of episodes over batches is the more efficient memory footprint.
+For example, an algorithm like DQN needs to have both observations and
+next observations (to compute the TD error-based loss) in the train batch, thereby duplicating an
+already large observation tensor. Using episode objects for most of the time reduces the memory need
+to a single observation-track, which contains all observations, from reset to terminal.
+
+This page explains in detail what working with RLlib's Episode APIs looks like.
+
+SingleAgentEpisode
+==================
+
+This page describes the single-agent case only.
+See here for a detailed description of the multi-agent case (work in progress).
+
+
+Creating a SingleAgentEpisode
+-----------------------------
+
+RLlib usually takes care of creating :py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode` instances
+and moving them around, for example from :py:class:`~ray.rllib.env.env_runner.EnvRunner` to :py:class:`~ray.rllib.core.learner.learner.Learner`.
+However, here is how to manually generate and fill an initially empty episode with dummy data:
+
+.. literalinclude:: doc_code/sa_episode.py
+    :language: python
+    :start-after: rllib-sa-episode-01-begin
+    :end-before: rllib-sa-episode-01-end
+
+The :py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode` constructed and filled preceding should roughly look like this now:
+
+.. figure:: images/episodes/sa_episode.svg
+    :width: 750
+    :align: left
+
+    **(Single-agent) Episode**: The episode starts with a single observation (the "reset observation"), then
+    continues on each timestep with a 3-tuple of `(observation, action, reward)`. Note that because of the reset observation,
+    every episode - at each timestep - always contains one more observation than it contains actions or rewards.
+    Important additional properties of an Episode are its `id_` (str) and `terminated/truncated` (bool) flags.
+    See further below for a detailed description of the :py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode`
+    APIs exposed to the user.
+
+
+Using the getter APIs of SingleAgentEpisode
+-------------------------------------------
+
+Now that there is a :py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode` to work with, one can explore
+and extract information from this episode using its different "getter" methods:
+
+.. figure:: images/episodes/sa_episode_getters.svg
+    :width: 750
+    :align: left
+
+    **SingleAgentEpisode getter APIs**: "getter" methods exist for all of the Episode's fields, which are `observations`,
+    `actions`, `rewards`, `infos`, and `extra_model_outputs`. For simplicity, only the getters for observations, actions, and rewards
```

Comment:
We should mention that the getter of `extra_model_outputs`  works a bit differently - this is specifically useful to know, if all extra-model-outputs should be extracted for a single timestep (we have then to iterate through the keys)

Reply:
Added more context and a link to the `get_extra_model_outputs` docstring.